e2e/machineset_migration_test.go

```diff
@@ -46,180 +52,586 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
+					}
+
+					return true, nil
+				}, capiframework.WaitMedium, capiframework.RetryMedium).Should(BeTrue(), "New InfraTemplate should be created with different name")
+
+				By("Verifying new InfraTemplate has the updated InstanceType")
+				newAWSMachineTemplate, err := capiframework.GetAWSMachineTemplateByPrefix(cl, mapiMSAuthMAPIName, capiframework.CAPINamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to get new awsMachineTemplate  %s", newAWSMachineTemplate)
+				Expect(newAWSMachineTemplate.Spec.Template.Spec.InstanceType).To(Equal(newInstanceType))
+
+				By("Verifying the old InfraTemplate is deleted")
+				verifyAWSMachineTemplateDeleted(originalAWSMachineTemplateName)
+			})
+		})
+
+		Context("when switching MAPI MachineSet spec.authoritativeAPI to ClusterAPI", Ordered, func() {
+			BeforeAll(func() {
+				updateMachineSetAuthoritativeAPI(mapiMachineSet, mapiv1beta1.MachineAuthorityClusterAPI, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifySynchronizedCondition(mapiMachineSet, mapiv1beta1.MachineAuthorityClusterAPI)
+			})
+
+			It("should be rejected when scaling MAPI MachineSet", func() {
+				By("Scaling up MAPI MachineSet to 1")
+				mapiframework.ScaleMachineSet(mapiMSAuthMAPIName, 1)
+
+				By("Verifying MAPI MachineSet replicas is restored to original value 0")
+				//verifyMAPIMachinesetReplicas(mapiMachineSet, 0)
+			})
+
+			It("should be rejected when when updating providerSpec of MAPI MachineSet", func() {
+				By("Getting the current MAPI MachineSet providerSpec InstanceType")
+				originalInstanceType := getMAPIMachineSetInstanceType(ctx, cl, mapiMSAuthMAPIName)
+
+				By("Updating the MAPI MachineSet providerSpec InstanceType")
+				updateMAPIMachineSetInstanceType(ctx, cl, mapiMachineSet, "m5.xlarge")
+
+				By("Verifying MAPI MachineSet InstanceType is restored to original value")
+				expectMAPIMachineSetInstanceType(ctx, cl, mapiMSAuthMAPIName, originalInstanceType)
+			})
+
+			It("should update MAPI MachineSet and remove old InfraTemplate when CAPI MachineSet points to new InfraTemplate", func() {
+				By("Creating a new awsMachineTemplate with different spec")
+				newInstanceType := "m6.xlarge"
+				originalAWSMachineTemplateName := capiMachineSet.Spec.Template.Spec.InfrastructureRef.Name
+				newAWSMachineTemplate = createAWSMachineTemplateWithInstanceType(ctx, cl, originalAWSMachineTemplateName, newInstanceType)
+
+				By("Updating CAPI MachineSet to point to the new InfraTemplate")
+				updateCAPIMachineSetInfraTemplate(capiMachineSet, newAWSMachineTemplate.Name)
+
+				By("Verifying the old InfraTemplate is deleted")
+				// bug https://issues.redhat.com/browse/OCPBUGS-61103
+				//verifyAWSMachineTemplateDeleted(originalAWSMachineTemplateName)
+
+				By("Verifying the MAPI MachineSet is updated to reflect the new template")
+				Eventually(func() string {
+					mapiMachineSet, _ = mapiframework.GetMachineSet(ctx, cl, mapiMSAuthMAPIName)
```

Comment:
same as above, we can call `mapiframework.GetMachineSet(ctx, cl, mapiMSAuthMAPIName)` once, outside the `Eventually()`, then use `k.Object` and standard matchers.